asyncio_mqtt/client.py

```diff
@@ -87,6 +87,14 @@ def __init__(
         bind_port: int = 0,
         clean_start: bool = mqtt.MQTT_CLEAN_START_FIRST_ONLY,
         properties: Optional[Properties] = None,
+        message_retry_set: int = 20,
+        # By default, in order to optimize the transmission of small data packets,
+        # the send buffer size is set to 2048 bytes. To transfer large data packets (>50MB),
+        # it is recommended to increase the buffer value.
+        # It is also possible to set all possible socket settings.
+        # See https://docs.python.org/3/library/socket.html#socket.socket.setsockopt for more information.
+        # If the socket_options is None, then the system defaults are used.
+        socket_options: Optional[Tuple[int, int, Union[int, bytes]]] = (socket.SOL_SOCKET, socket.SO_SNDBUF, 2048),
```

Comment:
Good job with the initial typing here. :+1: I really like the details in the comment as well

I suggest that we drop the default `SO_SNDBUF` setting. Move it to the comment, though, as a nice example.

I also suggest that we expand on this, so that people can set multiple options using any of [the three overloads of setsockopt](https://docs.python.org/3/library/socket.html#socket.socket.setsockopt). First, we define:
```python
# See the overloads of `socket.setsockopt` for details.
SocketOption = Union[
    Tuple[int, int, Union[int, bytes]],
    Tuple[int, int, None, int],
]
```

Second, we accept an iterable of socket options in the `Client.__init__`.
```python
socket_options: Iterable[SocketOption],
```

Reply:
Yes, is good idea!

Reply:
I found it worth adding one more thing so that there would be no compatibility issues with the new library version when upgrading.
If we do this: `socket_options: Iterable[SocketOption],` then we get an error
**TypeError: __init__() missing 1 required keyword-only argument: 'socket_options'**, because we have no input parameter when constructing an object.

Maybe add a default value?
For example: `socket_options: Optional[Iterable[SocketOption]] = ()`